src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -217,26 +351,47 @@ async def _schedule_rpc_coro(object rpc_coro,
 
 
 async def _handle_rpc(list generic_handlers, RPCState rpc_state, object loop):
```

Comment:
Correct me if I'm wrong, but it looks like this entire function could live in the Python layer instead of the Cython layer.

Reply:
Yes. We could move it to Python layer, the `RPCState` Cython class should be automatically recognized when crossing the border of Cython / Python, so we are not losing access to C/C++ objects.

Reply:
My rationale *was* that we shouldn't lose any performance. I'm wondering where the 10% performance benefit comes from if all of these are Python types. What's happening in Cython should happen basically the same way it would run as Python bytecode:
 - The GIL will remain held
 - The various `PyObject`s will be interacted with the same way

I suppose the difference is that instead of translating Python bytecode to interactions with the `PyObject`s, Cython functions directly make these calls, with no conditionals.

Thanks for looking into the performance delta!